store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/AbstractCoreKeyAccumuloElementConverter.java

```diff
@@ -260,6 +271,31 @@ public byte[] buildColumnQualifier(final String group, final Properties properti
         return stream.toByteArray();
     }
 
+    /**
+     * Populates the set of groups that aggregate with a time sensitive
+     * aggregation function.
+     */
+    private void populateTimeSensitiveGroups() {
+        // Find the groups that use a time sensitive aggregation function
+        for (final String group : aggregatedGroups) {
+            // Check all type definitions for the group
+            Stream<TypeDefinition> typeStream = StreamSupport.stream(
+                schema.getElement(group).getPropertyTypeDefs().spliterator(), true);
+            // Check if a time sensitive function is used anywhere
+            boolean timeSensitiveGroup = typeStream.anyMatch(td -> {
+                if (td.getAggregateFunction() == null) {
+                    return false;
+                }
+                return AccumuloStoreConstants.TIME_SENSITIVE_AGGREGATORS.contains(
+                    td.getAggregateFunction().getClass().getSimpleName());
+            });
```

Comment:
Single return condition. please

```suggestion
            boolean timeSensitiveGroup = typeStream.anyMatch(td ->
                    td.getAggregateFunction() != null
                            && AccumuloStoreConstants.TIME_SENSITIVE_AGGREGATORS
                            .contains(td.getAggregateFunction().getClass().getSimpleName()));
```

Reply:
Also I don't think you want to use simple name.

Reply:
Using simple name as allows for easy comparison rather than messing around with instances of classes etc. I could use the full name if needed though.

Reply:
Else where in Gaffer we use the full names.